api/src/scripts/populate_db.py

```diff
@@ -223,35 +187,82 @@ def add_entity(entity, priority):
+                target_feed = self.query_feed_by_stable_id(target_stable_id, None)
+                if not target_feed:
+                    self.logger.warning(f"Could not find redirect target feed {target_stable_id} for feed {stable_id}")
+                    continue
 
-                if target_feed:
-                    if target_feed.id != feed.id and (feed.id, target_feed.id) not in redirects_set:
-                        redirect = Redirectingid(source_id=feed.id, target_id=target_feed.id, redirect_comment=comment)
-                        add_entity(redirect, 5)
-                    else:
-                        self.logger.error(f"Feed has redirect pointing to itself {mdb_id}")
+                if feed.id == target_feed.id:
+                    self.logger.error(f"Feed has redirect pointing to itself {stable_id}")
                 else:
-                    self.logger.warning(f"Could not find redirect target feed {target_stable_id} for feed {mdb_id}")
-
-        priority = 1
-        while not entities_index.empty():
-            next_priority, entity_index = entities_index.get()
-            if priority != next_priority:
-                self.logger.debug(f"Flushing for priority {priority}")
-                priority = next_priority
-                self.db.flush()
-            self.fast_merge(entities[entity_index])
-
-        self.logger.info("Refreshing MATERIALIZED FEED SEARCH VIEW - Started")
-        self.db.session.execute(text(f"REFRESH MATERIALIZED VIEW CONCURRENTLY {t_feedsearch.name}"))
-        self.logger.info("Refreshing MATERIALIZED FEED SEARCH VIEW - Completed")
-
-        self.db.commit()
+                    if all(redirect.target_id != target_feed.id for redirect in feed.redirectingids):
+                        feed.redirectingids.append(
+                            Redirectingid(source_id=feed.id, target_id=target_feed.id, redirect_comment=comment)
+                        )
+                        # Flush to avoid FK violation
+                        self.db.session.flush()
+
+    def populate_db(self):
+        """
+        Populate the database with the sources.csv data
+        """
+        self.logger.info("Populating the database with sources.csv data")
+        for index, row in self.df.iterrows():
+            self.logger.debug(f"Populating Database with Feed [stable_id = {row['mdb_source_id']}]")
+            # Create or update the GTFS feed
+            data_type = self.get_data_type(row)
+            stable_id = self.get_stable_id(row)
+            feed = self.query_feed_by_stable_id(stable_id, data_type)
+            if feed:
+                self.logger.debug(f"Updating {feed.__class__.__name__}: {stable_id}")
+            else:
+                self.logger.info(f"Creating {feed.__class__.__name__}: {stable_id}")
+                feed = self.get_model(data_type)(id=generate_unique_id(), data_type=data_type, stable_id=stable_id)
```

Comment:
Feed is `None` at line 220 so the class would be `NoneType`:
```suggestion
                feed = self.get_model(data_type)(id=generate_unique_id(), data_type=data_type, stable_id=stable_id)
                self.logger.info(f"Creating {feed.__class__.__name__}: {stable_id}")
```

Reply:
Nice catch! Thanks!